app/controllers/fed_ax_api_controller.rb

```diff
@@ -0,0 +1,128 @@
+require "#{ Rails.root }/lib/shipping_api.rb"
+
+class FedAxApiController < ApplicationController
+  include FedAxApiWrapper
+
+  def quote
+    begin
+      content = ShippingApi.query(shipping_quote_params)
+
+      # OPTIMIZE: is this the best way to handle response status from the API Wrapper?
+      if content[:status] == 200
+        status = :ok
+      else content[:status] == 204
+        status = :no_content
+      end
+
+      # handling for receiving just the shipping quote for a given service type
+      if params[:shipping]
+        carrier, service_type = shipping_choice_params
+        shipping_details = shipping_selection(content[:quotes], carrier, service_type)
+
+        content = {}
+        content[:quote] = shipping_details
+        content[:status] = 200
+
+        status = :ok
+      end
+
+    # this is not a hash rocket. it is doing some magic thing where it's assigning
+    # the error object to a variable. now we can access the error object's message
+    rescue ActionController::ParameterMissing => error # this is raised by missing params
+      content = { message: "#{ error.message.capitalize }" }
+      status = :bad_request
+    rescue ActiveShipping::ResponseError => error # this error is raised by bad address formats
+      content = { message: "#{ error.message }" }
+      status = :bad_request
+    end
+
+    render json: content, status: status
+  end
+
+
+  def ship
+    begin
+      content = ShippingApi.query(shipping_quote_params)
+
+      # OPTIMIZE: is this the best way to handle response status from the API Wrapper?
+      if content[:status] == 200
+        carrier, service_type = shipping_choice_params
+        shipping_details = shipping_selection(content[:quotes], carrier, service_type)
+        log_entry = ApiResponse.create(shipping_details)
+        log_entry.log!(save_shipping_quote_params)
+
+        content = {}
+        content[:receipt] = shipping_details
+        content[:status] = 201
+
+        status = :created
+      else content[:status] == 204
+        status = :no_content
+      end
+
+
+    # this is not a hash rocket. it is doing some magic thing where it's assigning
+    # the error object to a variable. now we can access the error object's message
+    rescue ActionController::ParameterMissing => error
+      content = { message: "#{ error.message.capitalize }" }
+      status = :bad_request
+    rescue ActiveShipping::ResponseError => error
+      content = { message: "#{ error.message }" }
+      status = :bad_request
+    end
```

Comment:
Lines 29-37 are exactly the same as those directly above. Prime candidate for pulling out into their own method.